umbraco/Umbraco-CMS
# V13: Webhook all the things

### Prerequisites


If there's an existing issue for this PR then this fixes 

### Description
This adds in more WebHooks into the CMS as the are plenty of Notifications/Events raised by Umbraco that people may want to be notified by as a webhook posted externally to a third party system.

This PR adds in a lot of WebHook Event Types but they are not automatically added by default and would require you to update the Webhook Collection in a composer like so, where an implementor can pick and choose what collection of WebHooks they want to add.

```csharp
// Pick and choose from the collection of Webhooks
builder.WebhookEvents()
            .AddContentWebhooks()
            .AddDataTypeWebhooks()
            .AddDictionaryWebhooks()
            .AddDomainWebhooks()
            .AddLanguageWebhooks()
            .AddMediaWebhooks()
            .AddMemberWebhooks()
            .AddMemberTypeWebhooks()
            .AddPackageWebhooks()
            .AddPublicAccessWebhooks()
            .AddRelationWebhooks()
            .AddScriptWebhooks()
            .AddStylesheetWebhooks()
            .AddTemplateWebhooks()
            .AddUserWebhooks();
            
// Or go nuts and add them all
builder.WebhookEvents()
            .AddAllAvailableWebhooks();
```


## Test Notes
* Enable a webhook from the long list and ensure the correct information is sent
* Use a site such as https://webhook.site to test the payloads

## Updated
### Content
- [x] ContentCopied 
- [x] ContentDeletedBlueprint
- [x] ContentDeletedVerions
- [x] ContentDeleted
- [x] ContentEmptiedRecycleBin
- [x] ContentMovedToRecycleBin
- [x] ContentMoved
- [x] ContentPublished
- [x] ContentRolledBack
- [x] ContentSavedBluePrint
- [x] ContentSaved
- [x] ContentSorted
- [x] ContentUnpublished

### DataType
- [x] DataTypeDeleted
- [x] DataTypeMoved
- [x] DataTypeSaved

### Dictionary
- [x] DictionaryItemDeleted
- [x] DictionaryItemSaved

### Domain
- [x] DomainDeleted
- [x] DomainSaved

### Language
- [x] LanguageDeleted
- [x] LanguageSaved

### Media
- [x] MediaDeleted
- [x] MediaEmptiedRecycleBin
- [x] MediaMovedToRecycleBin
- [x] MediaMoved
- [x] MediaSaved

### MediaType
- [x] MediaTypeChanged
- [x] MediaTypeDeleted
- [x] MediaTypeMoved
- [x] MediaTypeSaved

### Member
- [x] MemberDeleted
- [x] MemberSaved
- [x] MemberGroupDeleted
- [x] MemberGroupSaved
- [x] AssignedMemberRoles
- [x] RemovedMemberRoles
- [x] ExportedMember

### Member Type
- [x] MemberTypeSaved
- [x] MemberTypeDeleted
- [x] MemberTypeChanged
- [x] MemberTypeMoved

### Package
- [x] ImportedPackage

### Public Access
- [x] PublicAccessEntryDeleted
- [x] PublicAccessEntrySaved

### Relation
- [x] RelationDeleted
- [x] RelationSaved
- [x] RelationTypeDeleted
- [x] RelationTypeSaved

### Script
- [x] ScriptDeleted
- [x] ScriptSaved

### Stylesheet
- [x] StylesheetDeleted
- [x] StylesheetSaved

### Template
- [x] PartialViewDeleted
- [x] PartialViewSaved
- [x] TemplateDeleted
- [x] TemplateSaved

### User
- [x] UserDeleted
- [x] UserSaved
- [x] UserLocked
- [x] UserUnlocked
- [x] UserLoginFailed
- [x] UserLoginSuccess
- [x] UserLogoutSuccess
- [x] UserGroupDeleted
- [x] UserGroupSaved
- [x] UserPasswordChanged
- [x] UserPasswordReset
- [x] UseForgottonPasswordRequested
- [x] UserForgottonPasswordChanged
- [x] UserLoginRequiresVerification
- [x] UserTwoFactorRequested
- [x] AssignedUserGroupPermissions
